### ISSUE TITLE
Add interpreter for BatchNormInferenceOp

### ISSUE THREAD

**Issue by sdasgup3** [openxla/stablehlo#963]
Following the [testing guidelines](https://github.com/openxla/stablehlo/blob/main/docs/interpreter_checklist.md)

**Comment by burmako** [comment]
Reopening per #1571.

**Comment by FengJungle** [comment]
Intepreter still dose not support eval BatchnormInference op

**Comment by ghpvnist** [comment]
Hi @FengJungle, there is a workaround for decomposing BatchNormInference op to run it on the interpreter:
```
// convert stablehlo to mhlo
mlir-hlo-opt --stablehlo-legalize-to-hlo batch_norm_inference.mlir

// convert mhlo to hlo
tf-mlir-translate -- --mlir-hlo-to-hlo-text batch_norm_inference.mlir

// decompose batch_norm_inference in HLO
hlo-expand --input_format=hlo --batch_norm_expander batch_norm_inference.hlo

// convert hlo to mhlo
tf-mlir-translate -- --hlo-text-to-mlir-hlo batch_norm_inference.hlo

// convert mhlo to stablehlo
mlir-hlo-opt --hlo-legalize-to-stablehlo batch_norm_inference.mlir
```

Unfortunately there is no way to run batch norm inference as a standalone op. However, a recommended workaround is to wrap this decomposed op into a composite if you need to keep this abstraction. For example:
```
func.func @batch_norm_inference_impl(%operand : tensor<2x2x2xf64>, %scale: tensor<2xf64>, %offset: tensor<2xf64>, %mean : tensor<2xf64>, %variance : tensor<2xf64>) -> tensor<2x2x2xf64> {
  // decomp goes here
  ...
  return %result : tensor<2x2x2xf64>
}

func.func @main() {
  ...
  %result = "stablehlo.composite"(%operand, %scale, %offset, %mean, %variance) {
    name = "stablehlo.batch_norm_inference",
    composite_attributes = {
        epsilon = 0.0 : f32,
        feature_index = 2 : i64
    },
    decomposition = @batch_norm_inference_impl,
    version = 0 : i32
  } : (tensor<2x2x2xf64>, tensor<2xf64>, tensor<2xf64>, tensor<2xf64>, tensor<2xf64>) -> tensor<2x2x2xf64>
  ...
}
```

**Comment by FengJungle** [comment]
@ghpvnist Thanks for your reply. I'll try the first method. 
For the second, I met another issue: how to lower stablehlo.composite op into linalg/llvm to run mlir-cpu-runner?  or to run stablehlo_backend in torch-mlir ? 
I tried to convert stablehlo.composite op into a call, still failed to run stablehlo_backend.
